lms/views/api/blackboard/files.py

```diff
@@ -18,36 +18,36 @@
     r"blackboard:\/\/content-resource\/(?P<file_id>[^\/]*)\/"
 )
 
-# The maxiumum number of paginated requests we'll make before returning.
-PAGINATION_MAX_REQUESTS = 25
-
-# The maximum number of results to request per paginated response.
-# 200 is the highest number that Blackboard will accept here.
-PAGINATION_LIMIT = 200
-
-
 @view_defaults(permission=Permissions.API, renderer="json")
 class BlackboardFilesAPIViews:
     def __init__(self, request):
         self.request = request
-        self.blackboard_api_client = request.find_service(name="blackboard_api_client")
+        self.blackboard = request.find_service(name="blackboard")
 
     @view_config(request_method="GET", route_name="blackboard_api.courses.files.list")
+    @view_config(
+        request_method="GET", route_name="blackboard_api.courses.files.list_folder"
+    )
     def list_files(self):
         """Return the list of files in the given course."""
+
         course_id = self.request.matchdict["course_id"]
+        folder_id = self.request.matchdict.get("folder_id")
 
-        files = []
-        path = f"courses/uuid:{course_id}/resources?type=file&limit={PAGINATION_LIMIT}"
+        results = self.blackboard.get_files(course_id, folder_id)
 
-        for _ in range(PAGINATION_MAX_REQUESTS):
-            response = self.blackboard_api_client.request("GET", path)
-            files.extend(BlackboardListFilesSchema(response).parse())
-            path = response.json().get("paging", {}).get("nextPage")
-            if not path:
-                break
+        for result in results:
+            if result["type"] == "File":
+                result["id"] = f"blackboard://content-resource/{result['id']}/"
+            elif result["type"] == "Folder":
+                result["parent_id"] = folder_id
+                result["contents"] = self.request.route_url(
+                    "blackboard_api.courses.files.list_folder",
+                    course_id=course_id,
+                    folder_id=result["id"],
+                )
```

Comment:
In all of the `js_config` stuff we return callbacks like this:

```python
"listFiles": {
                "authUrl": auth_url,
                "path": request.route_path(
                    "blackboard_api.courses.files.list", course_id=course_id
                ),
            },
```

I'm not sure if that's important in general, but it might make it easier if they are the same? Not sure. Might be worth asking the front-end guys because if we know we'll never want an `authURL` and they don't care it doesn't matter.